chain/chain/src/flat_storage_creator.rs

```diff
@@ -209,6 +242,7 @@ impl FlatStorageShardCreator {
                     info!(target: "store", %shard_id, %final_height, ?status, "Switching status to fetching state");
 
                     let mut store_update = chain_store.store().store_update();
+                    self.metrics.flat_head_height.set(final_head.height as i64);
```

Comment:
Not related to this PR, but I think there should be some high level comments for this function describing what it does at each state.

Reply:
Will do in next PR. I think we can also move code for each separate step to its own method because steps are independent.